sdk/typescript/src/signers/signer-with-provider.ts

```diff
@@ -89,40 +89,64 @@ export abstract class SignerWithProvider implements Signer {
   }
 
   /**
-   * Sign a transaction and submit to the Fullnode for execution. Only exists
-   * on Fullnode
+   * Sign a message using the keypair, with the `PersonalMessage` intent.
    */
-  async signAndExecuteTransaction(
+  async signMessage(message: Uint8Array): Promise<SignedMessage> {
+    const messageBytes = messageWithIntent(
+      IntentScope.PersonalMessage,
+      message,
+    );
+    const signature = await this.signData(messageBytes);
+    return {
+      messageBytes: toB64(messageBytes),
```

Comment:
dont think we need to care about the message bytes to return. the verifier should know what intent they are verifying to and construct this themselves. (to save communication cost too)

Reply:
Good call, removed and added `verifyMessage`.